ci-operator/step-registry/cucushift/installer/check/vsphere/disktype/cucushift-installer-check-vsphere-disktype-commands.sh

```diff
@@ -4,37 +4,39 @@ set -o nounset
 set -o errexit
 set -o pipefail
 
+# shellcheck source=/dev/null
 source "${SHARED_DIR}/govc.sh"
+# shellcheck source=/dev/null
 source "${SHARED_DIR}/vsphere_context.sh"
 export KUBECONFIG=${SHARED_DIR}/kubeconfig
 declare vsphere_datacenter
 declare vsphere_datastore
 unset SSL_CERT_FILE
 unset GOVC_TLS_CA_CERTS
-INFRA_ID=$(jq -r .infraID ${SHARED_DIR}/metadata.json)
+INFRA_ID=$(jq -r .infraID "${SHARED_DIR}"/metadata.json)
 check_result=0
 disk_type="${DISK_TYPE:-thin}"
 
 function check_disk_type() {
-    local ret=0 vm_lists disk_file vm_disk_type 
-    mapfile -t  vm_lists < <(govc ls /${vsphere_datacenter}/vm/${INFRA_ID})
-    if (( ${#vm_lists[@]} == 0 )); then
+    local ret=0 vm_lists disk_file vm_disk_type
+    mapfile -t vm_lists < <(govc find -name "$INFRA_ID*" -type m)
```

Comment:
The change is to work with UPI cluster ?

Reply:
Still debugging it, if have folder defined, this step will report error